schemas/2015-01-01/Sendgrid.Email.json

```diff
@@ -0,0 +1,115 @@
+{
+  "id": "http://schema.management.azure.com/schemas/2014-04-01-preview/Sendgrid.json",
+  "$schema": "http://json-schema.org/draft-04/schema#",
+  "title": "Sendgrid.Email",
+  "description": "Sendgrid Resource Types",
+  "resourceDefinitions": {
+    "accounts": {
+      "type": "object",
+      "properties": {
+        "name": {
```

Comment:
the limitation is fine, but usually customers would like to use an expression such as "[parameter('sendgridEmailName')]"  to make the template more flexible.
With the 4-50 chars check here, some long expression would fail and confuse customers. Two options to solve the problem:
1. Remove the length check here and leave it to service side.
2. use "anyOf": [
 { your name definition},
 { "$ref": "http://schema.management.azure.com/schemas/2015-01-01/deploymentTemplate.json#/definitions/expression"}
Please note we have to use "anyOf" here, as a short expression will match both string with 4-50 chars and expression, "oneOf" won't work here.